src/metabase/models/setting/multi_setting.clj

```diff
@@ -0,0 +1,74 @@
+(ns metabase.models.setting.multi-setting
+  (:require [metabase.models.setting :as setting]
+            [metabase.util.i18n :refer [tru]]))
+
+(defmulti dispatch-multi-setting
+  "Determine the dispatch value for a multi-Setting defined by `define-multi-setting`."
+  {:arglists '([setting-key])}
+  keyword)
+
+(defmulti get-multi-setting
+  "Get the value of a multi-Setting defined by `define-multi-setting` for the `impl` obtained by
+  calling `(dispatch-multi-setting setting-key)`."
+  {:arglists '([setting-key impl])}
+  (fn [setting-key impl]
+    [(keyword setting-key) (keyword impl)]))
+
+(defmulti set-multi-setting
+  "Update the value of a multi-Setting defined by `define-multi-setting` for the `impl` obtained by
+  calling `(dispatch-multi-setting setting-key)`."
+  {:arglists '([setting-key impl new-value])}
+  (fn [setting-key impl _]
+    [(keyword setting-key) (keyword impl)]))
```

Comment:
Are the use of `keyword`s here necessary? Seems like `setting-key` should always be a keyword, and `impl` could be more complex and flow through if we removed the `keyword` here. If these need to be keywords, perhaps just the appropriate asserts in the defining macros instead?